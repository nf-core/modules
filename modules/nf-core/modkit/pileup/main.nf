process MODKIT_PILEUP {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/ont-modkit:0.3.0--h5c23e0d_0':
        'biocontainers/ont-modkit:0.3.0--h5c23e0d_0' }"

    input:
    tuple val(meta), path(bam), path(bai)
    tuple val(meta2), path(fasta)
    tuple val(meta3), path(bed)

    output:
    tuple val(meta), path("*.bed")     , emit: bed     , optional: true
    tuple val(meta), path("*.bedgraph"), emit: bedgraph, optional: true
    tuple val(meta), path("*.log")     , emit: log     , optional: true
    path "versions.yml"                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args        = task.ext.args ?: ''
    def prefix      = task.ext.prefix ?: "${meta.id}"
    def reference   = fasta ? "--ref ${fasta}" : ""
    def include_bed = bed ? "--include-bed ${bed}" : ''

    """
    modkit \\
        pileup \\
        $args \\
        --threads ${task.cpus} \\
        --prefix ${prefix} \\
        $reference \\
        $include_bed \\
        $bam \\
        ${prefix}.tmp

    if test -d ${prefix}.tmp; then
        for file in ${prefix}.tmp/*; do
            if test -f \$file; then
                mv \$file \$(basename \$file)
            fi
        done
    else
        mv ${prefix}.tmp ${prefix}.bed
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        modkit: \$( modkit --version | sed 's/mod_kit //' )
    END_VERSIONS
    """

    stub:
    def args   = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.bed
    touch ${prefix}.bedgraph
    touch ${prefix}.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        modkit: \$( modkit --version | sed 's/mod_kit //' )
    END_VERSIONS
    """
}
