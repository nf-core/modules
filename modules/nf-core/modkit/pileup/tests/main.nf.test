nextflow_process {

    name "Test Process MODKIT_PILEUP"
    script "../main.nf"
    tag "modules"
    tag "modules_nfcore"
    tag "modkit"
    tag "modkit/pileup"
    process "MODKIT_PILEUP"

    test("[bam, bai], [], []") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true)
                ]
                input[1] = [[],[]]
                input[2] = [[],[]]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("[bam, bai], fasta, []") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true)
                ]
                input[1] = [
                    [ id: 'test_ref' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = [[],[]]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("[bam, bai], fasta, bed") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true)
                ]
                input[1] = [
                    [ id: 'test_ref' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = Channel.of('chr22\t0\t1000')
                    .collectFile(name: 'chr22.bed', newLine: true)
                    .map { file -> [ [ id:'chr22' ], file ] }
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("[bam, bai], fasta, bed - traditional") {

        config "./nextflow.traditional.config"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true)
                ]
                input[1] = [
                    [ id: 'test_ref' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = Channel.of('chr22\t0\t1000')
                    .collectFile(name: 'chr22.bed', newLine: true)
                    .map { file -> [ [ id:'chr22' ], file ] }
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("[bam, bai], fasta, bed - traditional, HP") {

        config "./nextflow.traditional_hp.config"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true)
                ]
                input[1] = [
                    [ id: 'test_ref' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = Channel.of('chr22\t0\t1000')
                    .collectFile(name: 'chr22.bed', newLine: true)
                    .map { file -> [ [ id:'chr22' ], file ] }
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("[bam, bai], fasta, bed - traditional, HP, bedgraph") {

        config "./nextflow.traditional_hp_bedgraph.config"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true)
                ]
                input[1] = [
                    [ id: 'test_ref' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = Channel.of('chr22\t0\t1000')
                    .collectFile(name: 'chr22.bed', newLine: true)
                    .map { file -> [ [ id:'chr22' ], file ] }
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("[bam, bai], [], [] - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true)
                ]
                input[1] = [[],[]]
                input[2] = [[],[]]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("[bam, bai], fasta, [] - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true)
                ]
                input[1] = [
                    [ id: 'test_ref' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = [[],[]]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("[bam, bai], fasta, bed - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true)
                ]
                input[1] = [
                    [ id: 'test_ref' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = Channel.of('chr22\t0\t1000')
                    .collectFile(name: 'chr22.bed', newLine: true)
                    .map { file -> [ [ id:'chr22' ], file ] }
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("[bam, bai], fasta, bed - traditional - stub") {

        config "./nextflow.traditional.config"
        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true)
                ]
                input[1] = [
                    [ id: 'test_ref' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = Channel.of('chr22\t0\t1000')
                    .collectFile(name: 'chr22.bed', newLine: true)
                    .map { file -> [ [ id:'chr22' ], file ] }
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("[bam, bai], fasta, bed - traditional, HP - stub") {

        config "./nextflow.traditional_hp.config"
        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true)
                ]
                input[1] = [
                    [ id: 'test_ref' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = Channel.of('chr22\t0\t1000')
                    .collectFile(name: 'chr22.bed', newLine: true)
                    .map { file -> [ [ id:'chr22' ], file ] }
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("[bam, bai], fasta, bed - traditional, HP, bedgraph - stub") {

        config "./nextflow.traditional_hp_bedgraph.config"
        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true)
                ]
                input[1] = [
                    [ id: 'test_ref' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = Channel.of('chr22\t0\t1000')
                    .collectFile(name: 'chr22.bed', newLine: true)
                    .map { file -> [ [ id:'chr22' ], file ] }
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}