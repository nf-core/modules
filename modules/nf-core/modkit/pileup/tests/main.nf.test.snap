{
    "[bam, bai], fasta, bed - traditional, HP": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_1.bed:md5,464a9870774c340753e79639aeaf76c2",
                            "test_2.bed:md5,118de4b653dd082d76faa8802df493eb"
                        ]
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_1.bed:md5,464a9870774c340753e79639aeaf76c2",
                            "test_2.bed:md5,118de4b653dd082d76faa8802df493eb"
                        ]
                    ]
                ],
                "bedgraph": [
                    
                ],
                "log": [
                    
                ],
                "versions": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T09:09:40.272533628"
    },
    "[bam, bai], fasta, bed - traditional, HP, bedgraph - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bedgraph": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T09:11:33.234594142"
    },
    "[bam, bai], fasta, []": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,f973de342df883efc1656c82a3a3978d"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,f973de342df883efc1656c82a3a3978d"
                    ]
                ],
                "bedgraph": [
                    
                ],
                "log": [
                    
                ],
                "versions": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T09:09:03.537667806"
    },
    "[bam, bai], fasta, bed - traditional - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bedgraph": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T09:10:45.47904997"
    },
    "[bam, bai], fasta, bed - traditional, HP - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bedgraph": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T09:11:06.023194848"
    },
    "[bam, bai], fasta, [] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bedgraph": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T09:10:17.995261822"
    },
    "[bam, bai], fasta, bed - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bedgraph": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T09:10:30.574897879"
    },
    "[bam, bai], [], []": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,f973de342df883efc1656c82a3a3978d"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,f973de342df883efc1656c82a3a3978d"
                    ]
                ],
                "bedgraph": [
                    
                ],
                "log": [
                    
                ],
                "versions": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T09:08:51.291346508"
    },
    "[bam, bai], [], [] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bedgraph": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T09:10:05.670301111"
    },
    "[bam, bai], fasta, bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,ac38ce3bed1f8aa770d1e1a9f332e170"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,ac38ce3bed1f8aa770d1e1a9f332e170"
                    ]
                ],
                "bedgraph": [
                    
                ],
                "log": [
                    
                ],
                "versions": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T09:09:15.042805699"
    },
    "[bam, bai], fasta, bed - traditional, HP, bedgraph": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_1_m_CG0_combined.bedgraph:md5,af55904c9acbafa17ff35ee3239152d0",
                            "test_2_m_CG0_combined.bedgraph:md5,13554927fb35b71a98c0e6bcdc3945c9"
                        ]
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ],
                "bed": [
                    
                ],
                "bedgraph": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_1_m_CG0_combined.bedgraph:md5,af55904c9acbafa17ff35ee3239152d0",
                            "test_2_m_CG0_combined.bedgraph:md5,13554927fb35b71a98c0e6bcdc3945c9"
                        ]
                    ]
                ],
                "log": [
                    
                ],
                "versions": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T09:09:52.968133216"
    },
    "[bam, bai], fasta, bed - traditional": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,db6333f714a8ea4aa33902404a8d4812"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,db6333f714a8ea4aa33902404a8d4812"
                    ]
                ],
                "bedgraph": [
                    
                ],
                "log": [
                    
                ],
                "versions": [
                    "versions.yml:md5,aa9f1fd8a0c9ab9709f381f864f6cdbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T09:09:27.735400826"
    }
}